pkg/apis/policy/validation/validation.go

```diff
@@ -118,6 +118,8 @@ func ValidatePodSecurityPolicySpec(spec *policy.PodSecurityPolicySpec, fldPath *
 	allErrs = append(allErrs, validatePSPDefaultAllowPrivilegeEscalation(fldPath.Child("defaultAllowPrivilegeEscalation"), spec.DefaultAllowPrivilegeEscalation, spec.AllowPrivilegeEscalation)...)
 	allErrs = append(allErrs, validatePSPAllowedHostPaths(fldPath.Child("allowedHostPaths"), spec.AllowedHostPaths)...)
 	allErrs = append(allErrs, validatePSPAllowedFlexVolumes(fldPath.Child("allowedFlexVolumes"), spec.AllowedFlexVolumes)...)
+	allErrs = append(allErrs, validatePodSecurityPolicySysctls(fldPath.Child("allowedUnsafeSysctls"), spec.AllowedUnsafeSysctls)...)
+	allErrs = append(allErrs, validatePodSecurityPolicySysctls(fldPath.Child("forbiddenSysctls"), spec.ForbiddenSysctls)...)
```

Comment:
same comment about validation here... we can't get more strict in the future. consider the following:

* validate format (already done, looks like)
* disallow empty items (already done, but error message could be improved... complains about format when a simple required error would be better)
* if the list contains "*", limit it to a single item (it's confusing to allow "*" and "foo"... the "*" covers "foo"
* disallow duplicate items
* don't allow something in the forbidden list to cover something in the allowed list. examples:
  * can't put `"*"` in forbiddenSysctls and `"foo"` in allowedUnsafeSysctls
  * can't put `"foo"` in forbiddenSysctls and `"foo"` in allowedUnsafeSysctls
  * can't put `"foo/*"` in forbiddenSysctls and `"foo/bar"` in allowedUnsafeSysctls

Reply:
Imagining the following scenario:
safe whitelist: `["kernel.shm_rmid_forced", "net.ipv4.ip_local_port_range", "net.ipv4.tcp_syncookies"]`
forbiddenSysctls: `["kernel.*"]`
allowedUnsafeSysctls: `["kernel.shmmax"]`

If the restrictions in the `forbiddenSysctls` apply only to the safe sysctls (meaning forbid all safe sysctls that are `kernel.` prefixed), it's perfectly fine to have `"kernel.*"` in `forbiddenSysctls` overlap the `"kernel.shmmax"` in `allowedUnsafeSysctls`.

However as @sttts suggets in https://github.com/kubernetes/kubernetes/pull/63717/files#r193043146, once the `kernel.shmmax` becomes safe, it gets forbidden by default cause the category changes. Which is not backward compatible.

Reply:
either
1. we change the semantics of `allowedUnsafeSysctls` to also apply to safe sysctls or
2. we disallow intersections of both lists (what @liggitt proposes above).

In both cases the incontinuity during upgrade goes away. 

With (1) we can have  `forbidden: "*", allowed: "foo"` to easily forbid all sysctls but the explicitly allowed ones. With (2) we cannot express that without knowing the whitelist.

Reply:
> If the restrictions in the forbiddenSysctls apply only to the safe sysctls (meaning forbid all safe sysctls that are kernel. prefixed), it's perfectly fine to have "kernel.*" in forbiddenSysctls overlap with the "kernel.shmmax" in allowedUnsafeSysctls.
> 
> However as @sttts suggets in https://github.com/kubernetes/kubernetes/pull/63717/files#r193043146, once the kernel.shmmax becomes safe, it gets forbidden by default cause the category changes. Which is not backward compatible.

The confusion of contradictory information ("`x.*` is forbidden" "`x.y` is allowed") and the incompatibility as things move from unsafe to safe (which we expect to happen) makes me think that isn't a great approach

Reply:
discussed with @sttts and @ingvagabund, and agreed on the following:

* forbidden means forbidden
* if something is matched by the forbiddenSysctls list, it is forbidden
* to prevent confusion, validation will prevent putting something in the allowedUnsafeSysctls list that is covered by something in the forbidden list
* evaluation will be done as described in https://github.com/kubernetes/kubernetes/pull/63717#discussion_r192778185 (matched by `forbiddenSysctls` forbids, internal "safe" list allows, matched by `allowedUnsafeSysctls` allows)

use cases this approach addresses well:
1a. forbidding all sysctls: `forbiddenSysctls:["*"]`
1b. forbidding a specific "safe" sysctl: `forbiddenSysctls:["kernel.shm_rmid_forced"]`
1c. allowing specific "unsafe" sysctls: `allowedUnsafeSysctls: ["foo"]`
1d. allowing glob-match "unsafe" sysctls: `allowedUnsafeSysctls: ["foo/*"]`
1e. allowing all "unsafe" sysctls: `allowedUnsafeSysctls: ["*"]`
1f. allowing a glob-match of "unsafe" sysctls, forbidding a specific one from that set (`allowedUnsafeSysctls:["foo/*"], forbiddenSysctls: ["foo/bar"]`)

use cases this approach accommodates, but not easily:
2a. forbidding all "safe" sysctls except for a whitelist (requires tracking changes to what kube considers safe and updating the `forbiddenSysctls` list in PSPs appropriately)

if the "forbid all safe except for a whitelist" use case proves to be important in the future, that could be accomodated with an `allowedSafeSysctls` field that would whitelist which safe sysctls were allowed. we don't expect that to be necessary, but that is a coherent option should we need it.